data/calculated_parser/functions.py

```diff
@@ -69,11 +70,293 @@ def sspeed(depth, latitude, temperature, salinity):
         press = [seawater.pres(d, latitude) for d in depth]
     except TypeError:
         press = seawater.pres(depth, latitude)
-
+    
     speed = seawater.svel(salinity, temperature, press)
     return np.array(speed)
 
 
+def sspeedmin(depth, lat, lon, temperature, salinity):
+    """
+    Finds the global minimum of the speed of sound
+
+    Parameters:
+    sspeed: Speed of Sound
+    latitude: The latitude(s) in degrees North
+    """
+    
+    speed = sspeed(depth, lat, temperature, salinity)
+    speed = speed.transpose()
+    
+    for x in range(speed.shape[0]):
+        for y in range(speed.shape[1]):
+            min_val = np.nanmin(speed[x,y])
+            idx = np.where(speed[x,y] == min_val)
+            if (np.isnan(min_val)):
+                speed[x,y] = np.nan
+            elif idx[0].shape[0] > 1:
+                idx = idx[0][0]
+
+                speed[x,y] = depth.values[idx]
+            else:
+                speed[x,y] = depth.values[idx]
+            #speed[x][y] = depth[np.where(speed[x][y] == np.nanmin(speed[x][y]))]  #np.nanmin(speed[x][y])
+    
+    speed = speed.transpose()
+
+    speed = speed[0]
+
+    return np.array(speed)
+
+def soniclayerdepth(depth, lat, lon, temperature, salinity):
+    """
+    Finds the local maxima of the speed of sound
+
+    Parameters:
+    sspeed: Speed of Sound
+    latitude: The latitude(s) in degrees North
+    """
+    
+    speed = sspeed(depth, lat, temperature, salinity)
+    speed = speed.transpose()
+    sld = np.nan
+    for x in range(speed.shape[0]):
+        for y in range(speed.shape[1]):
+            sca_value = np.nanmin(speed[x,y])
+            sca_idx = np.where(speed[x,y] == sca_value)
+            
+            if (np.isnan(sca_value)):
+                pass
+            else:
+                sca_idx = sca_idx[0][0]
+
+                subset = speed[x,y][0:int(sca_idx) + 1]
+                sld_value = subset.max()
+                
+                if (np.isnan(sld_value)):
+                    pass
+                else:
+                    sld_idx = np.where(subset == sld_value)[0][0]
+                    sld = depth.values[sld_idx]
+                    speed[x,y] = sld
+
+    speed = speed.transpose()
+    speed = speed[0]
+    return np.array(speed)
+
+
+def find_sca_idx(speed):
```

Comment:
I think it would help future users to include what `sca` means as part of the docstring.

Reply:
Fair enough. Also, based on your overall comment. This is a smaller function I made later to extract some of the work instead of rewriting it.
I can take this function and update the initial calculations I wrote without it

Reply:
Added to docstring for sca, sld, and cd